src/zenml/integrations/argilla/annotators/argilla_annotator.py

```diff
@@ -0,0 +1,298 @@
+            api_key = config.api_key
+            logger.debug("Using API key from settings.")
+
+        if api_key:
+            init_kwargs["api_key"] = api_key
+
+        if config.workspace is not None:
+            init_kwargs["workspace"] = config.workspace
+        if config.extra_headers is not None:
+            init_kwargs["extra_headers"] = json.loads(config.extra_headers)
+        if config.httpx_extra_kwargs is not None:
+            init_kwargs["httpx_extra_kwargs"] = json.loads(
+                config.httpx_extra_kwargs
+            )
+
+        try:
+            _ = rg.active_client()
+        except BaseClientError:
+            rg.init(**init_kwargs)
+        return rg.active_client()
+
+    def get_url_for_dataset(self, dataset_name: str) -> str:
+        """Gets the URL of the annotation interface for the given dataset.
+
+        Args:
+            dataset_name: The name of the dataset.
+
+        Returns:
+            The URL of the annotation interface.
+        """
+        dataset_id = self.get_dataset(dataset_name=dataset_name).id
+        return f"{self.get_url()}/dataset/{dataset_id}/annotation-mode"
+
+    def get_datasets(self) -> List[Any]:
+        """Gets the datasets currently available for annotation.
+
+        Returns:
+            A list of datasets.
+        """
+        old_datasets = self._get_client().list_datasets()
+        new_datasets = rg.FeedbackDataset.list()
+
+        # Deduplicate datasets based on their names
+        dataset_names = set()
+        deduplicated_datasets = []
+        for dataset in new_datasets + old_datasets:
+            if dataset.name not in dataset_names:
+                dataset_names.add(dataset.name)
+                deduplicated_datasets.append(dataset)
+
+        return deduplicated_datasets
+
+    def get_dataset_stats(self, dataset_name: str) -> Tuple[int, int]:
+        """Gets the statistics of the given dataset.
+
+        Args:
+            dataset_name: The name of the dataset.
+
+        Returns:
+            A tuple containing (labeled_task_count, unlabeled_task_count) for
+                the dataset.
+        """
+        dataset = self.get_dataset(dataset_name=dataset_name)
+        labeled_task_count = len(
+            dataset.filter_by(response_status="submitted")
+        )
+        unlabeled_task_count = len(
+            dataset.filter_by(response_status="pending")
+        )
+        return (labeled_task_count, unlabeled_task_count)
+
+    def add_dataset(self, **kwargs: Any) -> Any:
+        """Registers a dataset for annotation.
+
+        You must pass a `dataset_name` and a `dataset` object to this method.
+
+        Args:
+            **kwargs: Additional keyword arguments to pass to the Argilla
+                client.
+
+        Returns:
+            An Argilla dataset object.
+
+        Raises:
+            ValueError: if 'dataset_name' and 'dataset' aren't provided.
+        """
+        dataset_name = kwargs.get("dataset_name")
```

Comment:
While I am loving the fact that the overall implementation/abstraction for our annotators is coming together, I think the approach that we are using here with the keyword arguments is not very scalable. Especially with a high enough number of "required" parameters, the implementation can become very cluttered and we might end up having to write `kwargs.get("parameter_name")` multiple times like here. I briefly looked into possible solutions but unfortunately could not find a proper one myself right away. However, still wanted to write the comment here as a reminder.

Reply:
Yeah so my thought here was that I am also feeling the limits of this strategy. I think when I originally did `label_studio` we wanted to leave it open since I wasn't sure what the API would look like. Now that I've made 4 annotators, I think I have a bit more a solid feel for what the interface should look like, but I was thinking we can first merge these in and then I'd make a subsequent PR that would standardise / update the annotators API for all of them. I think that it can be improved, but I'd do that as a subsequent step. (And maybe I could do this so we merge them in just after a release, and then I'd make the fixes before the next release so that users aren't exposed to the API changes.

Reply:
That's a good idea. Can resolve this now.